### PR TITLE
Fixes #22892: declared compatibility in 7.2 is misleading

### PR DESCRIPTION
**Pull request by peckpeck**
https://issues.rudder.io/issues/22892

### CODE REVIEW

File: release-data/rudder-7.3.cson [hunk 1]
```diff
@@ -121,7 +121,7 @@ api-version = {
               }
 
 # Agent versions you can use with this version's server
-compatible-agents = [ "7.2", "7.3" ]
+compatible-agents = [ "6.2", "7.0", "7.1", "7.2", "7.3" ]
```
Comment (fanf) [inline release-data/rudder-7.3.cson:124]: 7.3 is only compatible with 7.1, 7.2 and 7.3

File: release-data/rudder-8.0.cson [hunk 2]
```diff
@@ -116,7 +116,7 @@ api-version = {
               }
 
 # Agent versions you can use with this version's server
-compatible-agents = [ "7.2", "7.3", "8.0" ]
+compatible-agents = [ "7.0", "7.1", "7.2", "7.3", "8.0" ]
```
Comment (fanf) [inline release-data/rudder-8.0.cson:119]: 8.0 will be only compatible 7.3 and 8.0 (and latter on 8.1)

**Comment by peckpeck** [comment]
PR updated with a new commit

**Review by fanf — APPROVED** [review]

**Comment by peckpeck** [comment]
OK, squash merging this PR

**Comment by peckpeck** [comment]
OK, merging this PR

**Comment by peckpeck** [comment]
OK, merging this PR

**Pull request merged**